pull_infra.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+
+def run(cmd, cwd=None, check=True):
+    print(f"🔧 Running: {' '.join(cmd)}")
+    result = subprocess.run(cmd, cwd=cwd, capture_output=True, text=True)
+    if check and result.returncode != 0:
+        print(f"❌ Error: {result.stderr.strip()}")
+        sys.exit(result.returncode)
+    return result.stdout.strip()
+
+
+def vendor_branch_exists(branch_name):
+    return branch_name in run(["git", "branch"], check=False)
```

Comment:
```suggestion
    return branch_name in run(["git", "branch", "--all"], check=False)
```

This is an actual bug because the function doesn't do what it says it does.

If the goal of the function is to determine if the *current* branch is the given branch, the current implementation works, but we need a new name for the function.

See:
https://git-scm.com/docs/git-branch